sdp.go

```diff
@@ -25,6 +25,8 @@ func trackDetailsFromSDP(log logging.LeveledLogger, s *sdp.SessionDescription) m
 	rtxRepairFlows := map[uint32]bool{}
 
 	for _, media := range s.MediaDescriptions {
+		tmpIncomingTracks := map[uint32]trackDetails{}
```

Comment:
Instead of `tmpIncomingTracks` would it be possible to filter+delete? Just to remove the extra variable declare, and so you don't need to do things like

```
						delete(incomingTracks, uint32(rtxRepairFlow))
￼						delete(tmpIncomingTracks, uint32(rtxRepairFlow))
```

Reply:
Hm, originally this code looked a lot nicer and there was no temporary map, but it broke one test for Plan B. If I remove the `tmpIncomingTracks`, I don't think I'd be able to filter out tracks for Plan B like:

```
m=video
a=ssrc:1 msid:label1 id1
a=ssrc:2 msid:label2 id2
a=recvonly
```

I could keep a set of track SSRCs to be kept, like `rtxRepairFlows` and then delete the rest from `incomingTracks` at the end. Would this work?

Reply:
This is what I was thinking about:

```diff
diff --git a/sdp.go b/sdp.go
index 071435c..807a2b5 100644
--- a/sdp.go
+++ b/sdp.go
@@ -25,7 +25,8 @@ func trackDetailsFromSDP(log logging.LeveledLogger, s *sdp.SessionDescription) m
 	rtxRepairFlows := map[uint32]bool{}
 
 	for _, media := range s.MediaDescriptions {
-		tmpIncomingTracks := map[uint32]trackDetails{}
+		// Plan B can have multiple tracks in a signle media section
+		tracksInCurrentMediaSection := map[uint32]struct{}{}
 		remoteIsSending := false
 		trackID := ""
 		trackLabel := ""
@@ -60,7 +61,6 @@ func trackDetailsFromSDP(log logging.LeveledLogger, s *sdp.SessionDescription) m
 						}
 						rtxRepairFlows[uint32(rtxRepairFlow)] = true
 						delete(incomingTracks, uint32(rtxRepairFlow)) // Remove if rtx was added as track before
-						delete(tmpIncomingTracks, uint32(rtxRepairFlow))
 					}
 				}
 
@@ -84,9 +84,6 @@ func trackDetailsFromSDP(log logging.LeveledLogger, s *sdp.SessionDescription) m
 				if rtxRepairFlow := rtxRepairFlows[uint32(ssrc)]; rtxRepairFlow {
 					continue // This ssrc is a RTX repair flow, ignore
 				}
-				if existingValues, ok := tmpIncomingTracks[uint32(ssrc)]; ok && existingValues.label != "" && existingValues.id != "" {
-					continue // This ssrc is already fully defined
-				}
 
 				if len(split) == 3 && strings.HasPrefix(split[1], "msid:") {
 					trackLabel = split[1][len("msid:"):]
@@ -95,14 +92,15 @@ func trackDetailsFromSDP(log logging.LeveledLogger, s *sdp.SessionDescription) m
 
 				// Plan B might send multiple a=ssrc lines under a single m= section. This is also why a single trackDetails{}
 				// is not defined at the top of the loop over s.MediaDescriptions.
-				tmpIncomingTracks[uint32(ssrc)] = trackDetails{codecType, trackLabel, trackID, uint32(ssrc)}
+				incomingTracks[uint32(ssrc)] = trackDetails{codecType, trackLabel, trackID, uint32(ssrc)}
+				tracksInCurrentMediaSection[uint32(ssrc)] = struct{}{}
 			}
 		}
 
-		for ssrc, incoming := range tmpIncomingTracks {
-			if remoteIsSending {
+		if !remoteIsSending {
+			for ssrc, _ := range tracksInCurrentMediaSection {
 				// Only include a=sendrecv or a=sendonly track details. In other words, filter out a=recvonly and a=inactive.
-				incomingTracks[ssrc] = incoming
+				delete(incomingTracks, ssrc)
 			}
 		}
 	}
```